From 1b9dd8a307e8df33d2ab50b9886af55b54391722 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:50:10 -0400
Subject: [ethernet] bnx2x: Use is_kdump_kernel() to detect kdump kernel

Message-id: <1409946615-5862-66-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91610
O-Subject: [RHEL7.1 PATCH 65/70] net/bnx2x: Use is_kdump_kernel() to detect kdump kernel
Bugzilla: 1059554 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1059554
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit c993189601fa89842eee96814e3ab2277fae3076
Author: Amir Vadai <amirv@mellanox.com>
Date:   Mon Aug 25 16:06:54 2014 +0300

    net/bnx2x: Use is_kdump_kernel() to detect kdump kernel

    Use is_kdump_kernel() to detect kdump kernel, instead of
    reset_devices.

    CC: Ariel Elior <ariel.elior@qlogic.com>
    CC: Michal Schmidt <mschmidt@redhat.com>
    Signed-off-by: Amir Vadai <amirv@mellanox.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
index ec5af12..ef811d1 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
@@ -21,6 +21,7 @@
 #include <linux/if_vlan.h>
 #include <linux/interrupt.h>
 #include <linux/ip.h>
+#include <linux/crash_dump.h>
 #include <net/tcp.h>
 #include <net/ipv6.h>
 #include <net/ip6_checksum.h>
@@ -64,7 +65,7 @@ static int bnx2x_calc_num_queues(struct bnx2x *bp)
  int nq = bnx2x_num_queues ? : netif_get_num_default_rss_queues();
 
  /* Reduce memory usage in kdump environment by using only one queue */
- if (reset_devices)
+ if (is_kdump_kernel())
   nq = 1;
 
  nq = clamp(nq, 1, BNX2X_MAX_QUEUES(bp));
diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
index 18c4cd9..4f310fa 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
@@ -41,6 +41,7 @@
 #include <linux/ethtool.h>
 #include <linux/mii.h>
 #include <linux/if_vlan.h>
+#include <linux/crash_dump.h>
 #include <net/ip.h>
 #include <net/ipv6.h>
 #include <net/tcp.h>
@@ -11956,7 +11957,7 @@ static int bnx2x_init_bp(struct bnx2x *bp)
  bp->disable_tpa = disable_tpa;
  bp->disable_tpa |= IS_MF_STORAGE_SD(bp) || IS_MF_FCOE_AFEX(bp);
  /* Reduce memory usage in kdump environment by disabling TPA */
- bp->disable_tpa |= reset_devices;
+ bp->disable_tpa |= is_kdump_kernel();
 
  /* Set TPA flags */
  if (bp->disable_tpa) {
-- 
1.7.1